ui/app/models/kv/metadata.js

```diff
@@ -77,6 +77,21 @@ export default class KvSecretMetadataModel extends Model {
     return array.reverse();
   }
 
+  // helps in long logic statements for state of a currentVersion
+  get currentSecret() {
+    const currentVersionObject = this.versions[this.currentVersion];
+    let state = 'created';
+    if (currentVersionObject.destroyed) {
+      state = 'destroyed';
+    } else if (currentVersionObject.deletion_time) {
+      state = 'deleted';
+    }
+    return {
+      state,
+      isDeactivated: state !== 'created',
+    };
+  }
```

Comment:
this reads a little easier to me
```suggestion
  get currentSecret() {
    const data = this.versions[this.currentVersion];
    const state = data.destroyed ? 'destroyed' : data.deletion_time ? 'deleted' : 'created';

    return {
      state,
      isDeactivated: state !== 'created',
    };
```

Reply:
I did not know you could chain like that! Thanks for TIL.